Add notes from Sept 2024 collab cafe (#744)
jupyterhub/team-compass: docs/meetings/collab-cafe/2024.md

# 2024 Collaboration Cafe Notes Archive

## 2024-09-17

### Check-in

- Min / @minrk / Simula
- Samuel Gaist / @sgaist / Idiap Research Institute (might be 5 min late)
- Sarah Gibson / @sgibson91 / 2i2c
- Erik Sundell / @consideratio / 2i2c
- Arnim Bleier /@arnim / GESIS
- Dan Sholler / @dsholler / Organizational Mycology
- shane knapp / @shaneknapp / UC Berkeley
- Kirstie Whitaker / @KirstieJane / Alan Turing Institute
- Raniere Silva / @rgaiacs / GESIS

### Celebrations and Shout-Outs :tada:

- Erik (@consideRatio) doing lots of maintenance and update on ldapauthenticator and others. Thank you!
- notebook 7 now on mybinder.org

## Agenda :clock2:

- **Min (10m)**: spreading meeting facilitation responsibilities: <https://github.com/jupyterhub/team-compass/issues/737>
  - Maybe bring it up on the forum
  - wider advertisement of meetings, e.g. forum. Most things are only on Issues right now

>   - relying on volunteers leads ot bias in who participates

Reviewer: ```suggestion
  - relying on volunteers leads to bias in who participates
```

  - Sarah offers mentorship for facilitators
  - Broader question: Standing agenda item on community engagement/activity, who is coming or not, etc.

>   - Nominate a person to make a list of where to share things (write here:

Reviewer: ```suggestion
  - Nominate a person to make a list of where to share things (write here):
```

    - Jupyter forum
    - Twitter? / Mastodon? Do we have those accounts anymore?
    - Gitter?
- **Dan Sholler (5 minutes)**: Voices of Jupyter Research Project with The Turing Way

>   - SoW here: <https://docs.google.com/document/d/1M2kfwvT7KHfgWd2Qph3BIM7seyrb8Lt23cVE_1HzvMw/edit#heading=h.x5kwcnqrf9m2>

Reviewer: ```suggestion
  - Scope of Work here: <https://docs.google.com/document/d/1M2kfwvT7KHfgWd2Qph3BIM7seyrb8Lt23cVE_1HzvMw/edit#heading=h.x5kwcnqrf9m2>
```

  - <https://github.com/jupyterhub/mybinder.org-deploy/issues/3087>
  - <https://github.com/jupyterhub/mybinder.org-deploy/issues/3083>
  - nebulous "issues" launching repos that don't have local images
  - local image cache limited to 2TB, if it needs to be pulled from docker hub lots of traffic and sometimes very slow
  - If pull fails due to timeout, this triggers a new build
  - short-term: Lower GESIS role in federation to avoid timeouts
  - "passthrough" registry docker configuration to reduce push/pull roundtrip all the way to docker hub
- **Shane (no idea how much time)**: Configurable HTTP proxy running out of ephemeral ports
  - happens when ~300+ users are logged and working simultaneously
  - massive user impact
  - <https://github.com/jupyterhub/configurable-http-proxy/issues/388#issuecomment-2347600588>
  - <https://blog.cloudflare.com/how-to-stop-running-out-of-ephemeral-ports-and-start-to-love-long-lived-connections/>
    - `The default ephemeral port range contains more than 28,000 ports (60999+1-32768=28232). Does that mean we can have at most 28,000 outgoing connections? That’s the core question of this blog post!`

>   - do we limit # of users per hub (will be 'difficult' for some course-specific hubs w/1200 users)?

Reviewer: ```suggestion
  - do we limit number of users per hub (will be 'difficult' for some course-specific hubs with 1200 users)?
```
